macleann/enchiridion-server
# EXPANDED SHOW SUPPORT!

# Major updates to allow searching for any show and adding any episode to users' playlists

Multiple updates across components that allow users to query TMDB shows, seasons, and episodes via `/seasons`, `/episodes`, and `/series` endpoints

### Changes to urls:
- Removed `/episodes` endpoints that will no longer be used with the new query structure

### Model changes:
- Added `series_name` field to `Episode` model

### Serializer changes:
- Added `series_name` field to `LocalEpisodeSerializer` serializer
- Added `first_air_date` field to `SeriesSerializer`
- Changes `seasons` field in `SeriesSerializer` to `required=False`

### Fixture changes:
- Added `series_name` field and values to each episode fixture

### Models updates:
- Reran `seed_db.sh` to update the migrations
- Adds `0002_episode_series_name.py` that updates the data relationships per seed script above

### Additions:

`series_view.py`:
- [Allow Any] permission classes set
- list and retrieve functions

### Bug fixes:

Fixes #21 Feature: Add a year field to the series that are returned
Fixes #22 Bug: Allow any user to search for series
Fixes #23 Feature: Rework seasons and episodes endpoints to include series_id parameter

## Type of change


# How to Test?

Make sure you've followed all installation and setup instructions in the README.

```
git fetch origin nm-searching-shows
git checkout nm-searching-shows
python manage.py runserver
```

- [ ] Open Postman
- [ ] For all following requests sent, make sure that there is no `Authorization: Token <token>` header
- [ ] All following requests should be successful
- [ ] Send a `GET` request to `http://localhost:8000/series?q=Regular Show`
- [ ] Send a `GET` request to `http://localhost:8000/series/84773`
- [ ] Send a `GET` request to `http://localhost:8000/seasons?series_id=84773`
- [ ] Send a `GET` request to `http://localhost:8000/seasons/1?series_id=84773`
- [ ] Send a `GET` request to `http://localhost:8000/episodes/tmdb_episodes?series_id=15260&season_number=1`
- [ ] Send a `GET` request to `http://localhost:8000/episodes/tmdb_single_episode?series_id=15260&season_number=3&episode_number=6`
- [ ] Send a `GET` request to `http://localhost:8000/playlists/1`, all episodes should have the `series_name` field on them

# Checklist:
